app/assets/stylesheets/shared/datatables.scss

```diff
@@ -265,4 +265,14 @@
       border-bottom: 1px solid darken($white, 10%);
     }
   }
+
+  @include media-breakpoint-down(md) {
+    div.dataTables_filter {
+      text-align: right;
+    }
+
+    .dt-buttons > .btn {
+      flex: unset;
+    }
+  }
```

Comment:
reset the flex shorthand so the select all checkbox isn't filling the width of the mobile viewport